sa/sa.go

```diff
@@ -1328,3 +1287,117 @@ func (ssa *SQLStorageAuthority) GetOrder(ctx context.Context, req *sapb.OrderReq
 
 	return order, nil
 }
+
+func (ssa *SQLStorageAuthority) getAuthorizations(ctx context.Context, table string, status string, registrationID int64, names []string, now time.Time) (map[string]*core.Authorization, error) {
+
+	if len(names) == 0 {
+		return nil, berrors.InternalServerError("no names received")
+	}
+
+	params := make([]interface{}, len(names))
+	qmarks := make([]string, len(names))
+	for i, name := range names {
+		id := core.AcmeIdentifier{Type: core.IdentifierDNS, Value: name}
+		idJSON, err := json.Marshal(id)
+		if err != nil {
+			return nil, err
+		}
+		params[i] = string(idJSON)
+		qmarks[i] = "?"
+	}
+
+	var auths []*core.Authorization
+	_, err := ssa.dbMap.Select(
+		&auths,
+		fmt.Sprintf(`SELECT %s FROM %s
+	WHERE registrationID = ? AND
+	expires > ? AND
+	identifier IN (%s) AND
+	status = '%s'`, authzFields, table, strings.Join(qmarks, ","), status),
```

Comment:
I think `status` can be a `?` here no? Is there a reason to use string interpolation except for the table name & variable number of `?`'s for the `IN`?